core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java

```diff
@@ -76,20 +80,32 @@ public class TableUpdateDetails implements Closeable {
         this.engine = engine;
         this.defaultColumnTypes = defaultColumnTypes;
         final int n = netIoJobs.length;
-        this.localDetailsArray = new ThreadLocalDetails[n];
-        for (int i = 0; i < n; i++) {
-            this.localDetailsArray[i] = new ThreadLocalDetails(
-                    configuration, netIoJobs[i].getUnusedSymbolCaches(), writer.getMetadata().getColumnCount());
-        }
         CairoConfiguration cairoConfiguration = engine.getConfiguration();
-        TableRecordMetadata tableMetadata = writer.getMetadata();
         this.millisecondClock = cairoConfiguration.getMillisecondClock();
         this.writerTickRowsCountMod = cairoConfiguration.getWriterTickRowsCountMod();
-        this.writer = writer;
+        this.defaultCommitInterval = configuration.getCommitIntervalDefault();
+        this.defaultMaxUncommittedRows = cairoConfiguration.getMaxUncommittedRows();
+        this.writerAPI = writer;
+        TableRecordMetadata tableMetadata = writer.getMetadata();
         this.timestampIndex = tableMetadata.getTimestampIndex();
-        this.tableNameUtf16 = writer.getTableName();
-        writer.updateCommitInterval(configuration.getCommitIntervalFraction(), configuration.getCommitIntervalDefault());
-        this.nextCommitTime = millisecondClock.getTicks() + writer.getCommitInterval();
+        this.tableNameUtf16 = Chars.toString(writer.getTableName());
+        if (writer instanceof TableWriterSPI) {
```

Comment:
is this WAL, non-WAL check ?

Reply:
could we use standard method of checking? `metadata.isWalEnabled()` ?

Reply:
The next line casts `writer` to `TableWriterSPI`, so while we could add `isWalEnabled()` method (which would be unused for now) to the metadata, the current code is less fragile to future changes. WDYT?

Reply:
nah, lets do this later

Reply:
Sounds good.